### ISSUE TITLE
Wireless not permanently enabled when turning on tcpip.

### ISSUE THREAD

**Issue by robertsaron** [Genymobile/scrcpy#1548]
- [ ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: [e.g. Debian, Windows, macOS...]
 - scrcpy version: [e.g. 1.12.1]
 - installation method: [e.g. manual build, apt, snap, brew, Windows release...]
 - device model:
 - Android version: [e.g. 10]

**Describe the bug**

On errors, please provide the output of the console (and `adb logcat` if relevant).

```
Please paste terminal output in a code block.
```

Please do not post screenshots of your terminal, just post the content as text instead.

In a previous version of scrcpy, once wireless was enabled it would remain on. In version 1.14 I have to turn on the wireless option of scrcpy, or plug it in all the time to a USB connection.

I am on the latest version of android, and windows 10 Pro.

**Comment by robertsaron** [comment]
Information from Terminal: 

INFO: scrcpy 1.14 <https://github.com/Genymobile/scrcpy>
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1
Press any key to continue...

When connected via usb, it works fine.

**Comment by rom1v** [comment]
> In version 1.14 I have to turn on the wireless option of scrcpy

scrcpy has no "wireless option". It just uses `adb`, which can connect to the device over USB or TCP/IP via `adb connect`.

How did you connect to your device wirelessly via `adb`?

**Comment by robertsaron** [comment]
Yes I connected via adb over tcpip. In a previous version I could enable it once, and never connect my phone to my computers USB again. Today, after turning on my computer, I am not able to connect to my device over tcpip, which I will refer to as wireless, wifi, or wirelessy going forward.

I have to plug my phone into a USB port, after every reboot.

**Comment by rom1v** [comment]
After every reboot of the device or the computer? (`adb tcpip` does not survive device reboot)

Anyway, it's related to `adb` behavior, not scrcpy. The adb version changes between scrcpy releases, check a previous version to be sure.

**Comment by mjjauhar** [comment]
If you are on windows. Try creating a batch file, and open that batch file instead of scrcpy.exe.

[How to create and enable batch file in Windows](https://www.windowscentral.com/how-create-and-run-batch-file-windows-10) 

Here is what I put in my batch file:

@ECHO OFF 
:: Batch file for scrcpy wireless
TITLE scrcpy wireless
ECHO >> Make sure you enabled adb debugging on your device.
ECHO >> Plug your device via usb.
ECHO >> Connect the device to the same Wi-Fi as your computer.
:: step 1: Enable adb over TCP/IP on your device: adb tcpip 5555
ECHO =======================================
ECHO Enabling adb over TCP/IP on your device
ECHO =======================================
adb tcpip 5555
PAUSE
ECHO >> Unplug your device
:: step 2: Connect to your device: adb connect DEVICE_IP:5555 (replace DEVICE_IP).
ECHO =======================================
ECHO Connecting wirelessly to your device
ECHO =======================================
adb connect DEVICE_IP:5555
:: step 3: Run scrcpy as usual.
ECHO =======================================
ECHO starting scrcpy wielessly.
ECHO =======================================
scrcpy --bit-rate 2M --max-size 800
:: It may be useful to decrease the bit-rate and the definition
PAUSE

**Comment by fusionneur** [comment]
To permanently enable adb via wifi that will persist upon reboot, login as root in termux and issue command:
setprop persist.adb.tcp.port 5555

No need to plug usb cable anymore afterwards.

**Comment by robertsaron** [comment]
@fusionneur Do I enable root through ADB on my android phone?  Or turn on Root in the ADB shello, on my windows computer? 
I have looked on line for both, and need some help on both. Will you please elaborate.

**Comment by fusionneur** [comment]
1. in termux:
su
setprop persist.adb.tcp.port 5555
setprop service.adb.tcp.port 5555
stop adbd
start adbd

2. Alternatively you can send the command with adb:
adb shell 'su --command "setprop persist.adb.tcp.port 5555"'
adb shell 'su --command "setprop service.adb.tcp.port 5555"'
adb tcpip 5555 #if you want to start right away
adb connect wlan0_ip:5555 #where wlan0_ip is the phone's wlan ip
scrcpy

**Comment by fusionneur** [comment]
I wrote a script to assist wireless connections.

https://github.com/fusionneur/ethernally

I tested in various scenarios and looks fine. If you find any problem, please open an issue. Thank you

**Comment by robertsaron** [comment]
@fusionneur  any way to get this script to work on windows? I tried to follow your steps on my device using termux, and got my device rooted. Termux, tells me no su program found on this device. I also tried @mjjauhar  batch file and that did not work either. So either I am doing something wrong, or wireless is just not for me.

**Comment by fusionneur** [comment]
you can run the shell script in windows through cygwin or windows subsystem for linux

Just one question.. you said "got my device rooted. Termux, tells me no su program found on this device. " but these 2 things contradicts each other. Are you really rooted? You should have magisk installed for being rooted and su should work in termux..

**Comment by robertsaron** [comment]
I do not have magisk installed. When Rooting my device it wiped everything and I did not know that would happen. So I am not sure if I want to proceed. How hard is it to get magisk installed?

**Comment by fusionneur** [comment]
rooting itself shouldn't wipe, only unlocking bootloader wipes it - which is however a prerequisite step for first time rooting. What device do you have? Some devices are more tricky to root than others. Personally I follow xda guides that are device specific when rooting.

**Comment by robertsaron** [comment]
I have the Pixel2 XL. And yeah this was my first time rooting it. I unlocked the bootloader.

**Comment by robertsaron** [comment]
I somehow got the batch file working that another user posted. I am good on the wireless for now. Though I would at somepoint like to finish the root process.

**Comment by fusionneur** [comment]
for rooting I suppose you could follow this guide: https://www.youtube.com/watch?v=aMPRRQu1hCk

**Comment by fusionneur** [comment]
I created this tool to ease wireless connectivity. Check it out and let me know of your experience.

https://github.com/fusionneur/ethernally
